### PR TITLE
fixed gcc 13 compiler warnings

### PR DESCRIPTION
**Pull request by bubbleguuum**
mostly function pointer casts and other benign warnings
one warning resulted in fixing inverted function parameters in g_demo.c

### CODE REVIEW

File: src/engine/r_drawlist.c [hunk 1]
```diff
@@ -120,7 +120,7 @@ void DL_ProcessDrawList(int tag, boolean(*procfunc)(vtxlist_t*, int*)) {
 			qsort(dl->list, dl->index, sizeof(vtxlist_t), SortDrawList);
 		}
 		else if (dl->index >= 2) {
-			qsort(dl->list, dl->index, sizeof(vtxlist_t), SortSprites);
+			qsort(dl->list, dl->index, sizeof(vtxlist_t), (__compar_fn_t)SortSprites);
```
Comment (Wolf3s) [inline src/engine/r_drawlist.c:123]: This struct  `__compar_fn_t` doesn´t exist

**Review by Wolf3s — CHANGES_REQUESTED** [review]
__compar_fn_t doesn´t exist

File: src/engine/r_drawlist.c (hunk 1 above)
Comment (Wolf3s) [inline src/engine/r_drawlist.c]: __compar_fn_t doesn´t exists.

**Review by Wolf3s — CHANGES_REQUESTED** [review]

File: src/engine/r_drawlist.c (hunk 1 above)
Comment (Wolf3s) [reply src/engine/r_drawlist.c]: i suggest removing this guy.

**Comment by bubbleguuum** [comment]
Ok I removed it although the proper cast to make the warning go away would have been `int (*)( const void*, const void* )`, which I admit is a bit ugly...

**Comment by Wolf3s** [comment]
> Ok I removed it although the proper cast to make the warning go away would have been `int (*)( const void*, const void* )`, which I admit is a bit ugly...

well the function qsort itself it´s ugly, but i don´t recommend casting like you did.

**Comment by Wolf3s** [comment]
also there´s more suggestions: remove on SC_Data (int (*)(void*, void*)) cast and add and replace scparser_t on sc_main.h like this:
````
typedef struct {
	const char* token;
	int64_t   ptroffset;
	char    type;
} scdatatable_t;

typedef struct {
	char    token[512];
	byte* buffer;
	char* pointer_start;
	char* pointer_end;
	int     linepos;
	int     rowpos;
	int     buffpos;
	int     buffsize;
	void (*open)(const char*);
	void (*close)(void);
	void (*compare)(const char*);
	int (*find)(boolean);
	char(*fgetchar)(void);
	void (*rewind)(void);
	char* (*getstring)(void);
	int (*getint)(void);
	int (*setdata)(byte*, const scdatatable_t*);
	int (*readtokens)(void);
	void (*error)(const char*);
} scparser_t;
````

**Comment by Wolf3s** [comment]
byte* and void* are similar.

**Comment by bubbleguuum** [comment]
> also there´s more suggestions: remove on SC_Data (int (_)(void_, void*)) cast and add and replace scparser_t on sc_main.h like 

Done.

**Comment by bubbleguuum** [comment]
> byte* and void* are similar.

indeed but GCC (by default, without special flags) issue warnings on pointer function assignments if the signatures do not exactly match. It is a good thing as it allows to catch errors when the function pointers are really not compatible (like the `g_demo.c` argument mismatch in the `D_MiniLoop` function call that this PR fixes)

All of the warnings fixed by this patch is for `gcc` invocation with no special arguments. So these are default warnings.

**Comment by Wolf3s** [comment]
merged as co-author.

**Pull request merged**